src/java/org/apache/cassandra/cql3/statements/DeleteStatement.java

```diff
@@ -201,8 +201,9 @@ protected ModificationStatement prepareInternal(ClientState state,
             {
                 checkFalse(stmt.isVirtual(), "DELETE statements must restrict all PRIMARY KEY columns with equality relations");
 
-                checkFalse(operations.appliesToRegularColumns(),
-                           "DELETE statements must restrict all PRIMARY KEY columns with equality relations in order to delete non static columns");
+                if (!operations.isEmpty())
```

Comment:
going to fork this off into its own ticket... should be fixed outside of accord first...